pkg/secureframe/api.go

```diff
@@ -0,0 +1,114 @@
+package secureframe
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/Jeffail/gabs/v2"
+)
+
+const (
+	restEndpoint = "https://api.secureframe.com"
+)
+
+// makeRequest performs an authenticated HTTP request to the specified endpoint and returns a JSON-friendly byte slice.
+func makeRequest(ctx context.Context, url, method, accessKey, secretKey string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("%s %s", accessKey, secretKey))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return nil, err
+	}
+
+	rb, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return rb, nil
+}
+
+// GetUsers returns a map of noncompliant users and related information.
+func GetUsers(ctx context.Context, accessKey, secretKey, types string) (map[string]map[string]string, error) {
+	// User types to consider as valid
+	requiredTypes := make(map[string]bool)
+	for _, t := range strings.Split(types, ",") {
+		requiredTypes[strings.ToLower(t)] = true
+	}
+
+	// Store a map containing [unique] user IDs and their respective attributes
+	users := make(map[string]map[string]string)
+
+	requestUrl := fmt.Sprintf("%s/users", restEndpoint)
+
+	rb, err := makeRequest(ctx, requestUrl, "GET", accessKey, secretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse response bytes using Gabs to avoid defining structs
```

Comment:
What do you think about using https://mholt.github.io/json-to-go/ to generate a Go struct from the JSON data? That would eliminate all the awkward data copying this code is doing to generate vars and then a map of strings - as well as eliminate an unusual code dependency.

Reply:
Updated in [`43b421d` (#4)](https://github.com/chainguard-dev/secureframe-policy-minder/pull/4/commits/43b421d147a39786d38ea0b32986443daa885a12).

I've used Gabs in the past but I agree that it's not great with all of the type assertions. The new way is definitely cleaner.